### ISSUE TITLE
panic: interface conversion: interface is float64, not int64

### ISSUE THREAD

**Issue by eanxgeek** [bndr/gojenkins#15]
This bit of code is resulting in this error:

```
builds, err := jenkins.GetAllBuildIds(jobName)

	if err != nil {
		panic(err)
	}
	fmt.Printf("Build ID\tTimeStamp\t\tResult\tIs Running\n")
	fmt.Println("------------------------------------------------------")
	for _, build := range builds {
		buildId := build.Number
		data, err := jenkins.GetBuild(jobName, buildId)

		if err != nil {
			panic(err)
		}
		upstreamBuild, _ := data.GetUpstreamBuild()
		fmt.Println(upstreamBuild)
...
```

> panic: interface conversion: interface is float64, not int64
> 
> goroutine 1 [running]:
> github.com/bndr/gojenkins.(*Build).GetUpstreamBuildNumber(0xc8200f68d0, 0xc820101ba0, 0x0, 0x0)
> 	/home/esammons/go/src/github.com/bndr/gojenkins/build.go:305 +0x13f
> github.com/bndr/gojenkins.(*Build).GetUpstreamBuild(0xc8200f68d0, 0x86e380, 0x0, 0x0)
> 	/home/esammons/go/src/github.com/bndr/gojenkins/build.go:317 +0x97
> main.main()
> 	/home/esammons/git/getjenkinsstats/getjenkinsstats.go:50 +0x407
> exit status 2

What  I am trying to do is see the results of the individual tests within a given build.  Our setup consists of the following:

Results Job
     -- Build starts upstream **project** build number **nn**
       
Actual tests run, pass, and fail are reported in the project; however, I have not found a way to get to that level of detail with this library.  When trying to dig into Upstream Projects I hit the error above.

Any thoughts on how I might get actual **test** results from a build?

**Comment by bndr** [comment]
This bug was fixed in latest commit: https://github.com/bndr/gojenkins/commit/268f0eafcc42f27430cea6efa28fb189b383e30c

By Test results you mean the console output? you can do build.GetConsoleOutput()

**Comment by eanxgeek** [comment]
But I was looking for something less verbose; for example in jenkins when I click on Test Results I get a table of All Tests w/ a columns; duration, fail, skip, pass, total.

Would it be possible to grab that without grabbing console output?

**Comment by bndr** [comment]
Then you probably need build.GetResultSet()

**Comment by eanxgeek** [comment]
I tried that and got back nothing, let me try again and see if I can determine what is going on.  Any thoughts, though, on the error I reported?

**Comment by bndr** [comment]
It should be fixed now.

**Comment by eanxgeek** [comment]
Thanks I'll pull down the update.

As for GetResultSet, here my url: http://jenkins.local.com:8080/job/dtests-jamq-700-candidate-tests-results/470/testReport/(root)/.

Here's my code:
```
for _, build := range builds {
		buildId := build.Number
		data, err := jenkins.GetBuild(jobName, buildId)

		if err != nil {
			panic(err)
		}

		//upstreamBuild, _ := data.GetUpstreamBuild()
		//fmt.Println(upstreamBuild)

		testResults, err := data.GetResultSet()

		if err != nil {
			fmt.Println(buildId, " : ", err)
		} else {
			fmt.Println(testResults)
		}
...
}
```

And the resulting stdout:

> Build ID	TimeStamp		Result	Is Running
> ------------------------------------------------------
> 470  :  No Result sets found
> 470   2016-02-23 12:00:58.407 -0500 EST   FAILURE   false

This repeats for every build that I have, Is there something I'm doing wrong?  BTW, Build 470 has over 260 tests, some pass and some fail.

**Comment by bndr** [comment]
It's possible there's a bug in GetResultSet() https://github.com/bndr/gojenkins/blob/master/build.go#L345

I'll need to take a look into it.

**Comment by bndr** [comment]
Hm.. Can you try this code?

```
...
for _, build := range builds {
        buildId := build.Number
        data, err := jenkins.GetBuild(jobName, buildId)
        
        if err != nil {
            panic(err)
        }

        //upstreamBuild, _ := data.GetUpstreamBuild()
        //fmt.Println(upstreamBuild)
        data.Poll()
        testResults, err := data.GetResultSet()

        if err != nil {
            fmt.Println(buildId, " : ", err)
        } else {
            fmt.Println(testResults)
        }
...
}
```

**Comment by eanxgeek** [comment]
I have created a small stub program to test this, Using your suggestion with data.Poll() I see no change.

```
package main


import (
		"fmt"
		"gojenkins"
		)

func main () {
	url := "http://locahost:8080/"
	jobName := "dtests-jamq-700-candidate-tests-results"

	jenkins, _ := gojenkins.CreateJenkins(url).Init()

	var buildId int64
	buildId = 433

	data, _ := jenkins.GetBuild(jobName, buildId)

	data.Poll()
	testResults, err := data.GetResultSet()

	if err != nil {
		fmt.Println(buildId, " : ", err)
	} else {
		fmt.Println(testResults)
	}
}
```

> $ go run stub.go 
> 433  :  No Result sets found

And I know this build has at least 60 tests that ran from an upstream builds.

With regards to the original float64 issue, this appears to be fixed.

**Comment by bndr** [comment]
Can you try it one more time? I just updated the library

**Comment by eanxgeek** [comment]
Looks good, I now have to parse and such but appears to be valid json output. -Thanks great help.

**Comment by bndr** [comment]
Some of properties are interfaces{} because anything can be there, here's the definition of the result object https://github.com/bndr/gojenkins/blob/master/build.go#L73

**Comment by bndr** [comment]
If you find any bugs or mistakes, feel free to open up a pull request.

**Comment by eanxgeek** [comment]
Great thanks!  Now this question may demonstrate just how new I am to Golang.  I tried to:
```
for result := testResults {
   fmt.Println(result)
}
```
I get an error that _cannot range over testResults (type *gojenkins.testResult)_ .  If you have thoughts, I'd appreciate them otherwise I'll continue to investigate.

**Comment by bndr** [comment]
It should be something like this. (Did not test this snippet)
```
resultSet, err := data.GetResultSet()
for _, suit := range resultSet.Suites {
   for _, case := range suit.Cases {
     fmt.Println(case)
   }
}

```

**Comment by eanxgeek** [comment]
Thanks I was actually getting there, I have the following and hit an error:
```
package main


import (
		"fmt"
		"gojenkins"
		)

func main () {
	url := "http://localhost:8080/"
	jobName := "dtests-jamq-700-candidate-tests-results"

	jenkins, _ := gojenkins.CreateJenkins(url).Init()

	var buildId int64
	buildId = 433

	data, _ := jenkins.GetBuild(jobName, buildId)

	data.Poll()
	testResults, err := data.GetResultSet()

	if err != nil {
		fmt.Println(buildId, " : ", err)
	} else {
		fmt.Println("Pass Count : ", testResults.PassCount)
		fmt.Println("Fail Count : ", testResults.FailCount)

		for suite := range testResults.Suites {
			fmt.Println(suite)
			for c := range suite.Cases {
				fmt.Println(c.Name)
			}
		}
	}
}
```

The error:
> ./stub.go:31: suite.Cases undefined (type int has no field or method Cases)

Your suggestion above will probably fix this.
